File: Memorize/MasterViewController.m
//
//  MasterViewController.m
//  Memorize
//
//  Created by Heather Shelley on 9/28/13.
//  Copyright (c) 2013 Mine. All rights reserved.
//

#import "MasterViewController.h"
#import "ReviewTabViewController.h"
#import "CardViewerTableViewController.h"

@interface MasterViewController ()

@end

@implementation MasterViewController

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([[segue identifier] isEqualToString:@"Normal"]) {
        [[segue destinationViewController] setReviewType:ReviewTypeNormal];
    } else if ([[segue identifier] isEqualToString:@"Reverse"]) {
        [[segue destinationViewController] setReviewType:ReviewTypeReverse];
    } else if ([[segue identifier] isEqualToString:@"Reviewing"]) {
        [[segue destinationViewController] setViewType:ViewTypeReviewing];
    } else if ([[segue identifier] isEqualToString:@"NotReviewing"]) {
        [[segue destinationViewController] setViewType:ViewTypeNotReviewing];
    }
}

@end

File: Memorize/MasterViewController.swift
//
//  MasterViewController.swift
//  Memorize
//
//  Created by Nick Shelley on 12/17/16.
//  Copyright © 2016 Mine. All rights reserved.
//

import UIKit

class MasterViewController: UITableViewController {
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        guard let idendifier = segue.identifier else { return }
        
        if let reviewViewController = segue.destination as? ReviewViewController {
            switch idendifier {
            case "NormalReview":
                reviewViewController.reviewType = .normal
            case "ReverseReview":
                reviewViewController.reviewType = .reverse
            default:
                break
            }
        }
        
        if let cardViewerController = segue.destination as? CardViewerTableViewController {
            switch idendifier {
            case "Reviewing":
                cardViewerController.viewType = .reviewing
            case "NotReviewing":
                cardViewerController.viewType = .notReviewing
            default:
                break
            }
        }

    }
    
}
